code/datums/components/speechmod.dm

```diff
@@ -51,6 +54,8 @@
 	var/message = speech_args[SPEECH_MESSAGE]
 	if(message[1] == "*")
 		return
+	if(!isnull(should_modify_speech) && !should_modify_speech?.Invoke(source, speech_args))
```

Comment:
```suggestion
	if(!isnull(should_modify_speech) && !should_modify_speech.Invoke(source, speech_args))
```
should_modify_speech being null has already been checked earlier in the same conditional.